pkg/controllers/repositoryserver/server.go

```diff
@@ -97,6 +97,32 @@ func (h *RepoServerHandler) getServerDetails(ctx context.Context) (string, strin
 }
 
 func (h *RepoServerHandler) checkServerStatus(ctx context.Context, serverAddress, username, password string) error {
+	fingerprint, err := kopia.ExtractFingerprintFromCertSecret(ctx, h.KubeCli, h.RepositoryServerSecrets.serverTLS.Name, h.RepositoryServer.Namespace)
+	if err != nil {
+		return errors.Wrap(err, "Failed to extract fingerprint from Kopia API server certificate secret data")
+	}
+	cmd := command.ServerStatus(
+		command.ServerStatusCommandArgs{
+			CommandArgs: &command.CommandArgs{
+				RepoPassword:   "",
+				ConfigFilePath: command.DefaultConfigFilePath,
+				LogDirectory:   command.DefaultLogDirectory,
+			},
+			ServerAddress:  serverAddress,
+			ServerUsername: username,
+			ServerPassword: password,
+			Fingerprint:    fingerprint,
+		})
```

Comment:
This code seem to be redundant in function checkServerTillCommandSucceed,see the line numbers- https://github.com/kanisterio/kanister/pull/2048/files#diff-8e1bee7434fbff0335bff585b3dda95d4c382937dc0e7086bb51bdf5c2738085R100-R115

Can we reuse this code there instead of writing it again

Reply:
Done !